include/irr/asset/IMeshPacker.h

```diff
@@ -20,10 +20,17 @@ namespace meshPackerUtil
         size_t MDIDataBuffMinAllocSize                       = 32ull;              /*  32B*/
     };
 
-    template <typename MeshBufferType>
+    template <typename BufferType>
     struct PackedMeshBuffer
     {
-        core::smart_refctd_ptr<MeshBufferType> packedMeshBuffer = nullptr;
+        //or output should look more like `return_type` from geometry creator?
```

Comment:
You don't need to return the smart pointers to any buffers, just offsets inside them because the buffers are always the same for all `commit`s and will be kept by the class.

`vertexInputParams` will be identical as well, so no need for a copy returned by every allocation.

Whatever is the same for all objects, you shall get from the factory that made them.

Reply:
`return_type` from geometry creator is overkill here, because you already know the different pipeline settings you will draw meshes with by the time you want to pack them.

geometry creator returns all this extra data because it tells you how to do it, packer wont need that.

Reply:
also because you dont need to return buffers, you don't need to template, and this struct can go into the base packer class (or a base of the base that is not templated upon the MDI struct)